plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1005,6 +1011,34 @@ def verify_size(self, location):
             with open(location) as fp:
                 verification.verify_size(fp, self.size)
 
+    def set_repodata(self, metadata_type, xml_snippet):
+        """
+        Compress metadata and put it into `repodata` attribute which will be saved to the db later.
+
+        :param metadata_type: key for the `repodata` dictionary which indicates type of metadata
+        :type  metadata_type: str
+        :param xml_snippet: utf-8 string which will be compressed and put into `repodata` dictionary
+        :type  xml_snippet: str
+        """
+        if metadata_type in constants.MANDATORY_METADATA_TYPES:
+            self.repodata[metadata_type] = bson.binary.Binary(gzip.zlib.compress(xml_snippet))
+        else:
+            self.repodata[metadata_type] = xml_snippet
```

Comment:
FWIW nothing has ever gone into this dictionary that isn't one of the three mandatory types, and there is no code in the importer that I know of that could possibly add something else. I wish it had been modeled as three separate fields on the document, instead of nesting them in a dictionary that always has exactly 3 members.

If it's helpful to assume that the dictionary will always have exactly the same three keys, feel free to do so.